src/core/src/CubicSpline.cpp

```diff
@@ -88,29 +103,43 @@ bool iDynTree::CubicSpline::computeCoefficients()
 
 bool iDynTree::CubicSpline::computeIntermediateVelocities()
 {
-    Eigen::MatrixXd A(m_velocities.size()-2, m_velocities.size()-2);
+    Eigen::SparseMatrix<double> A(m_velocities.size()-2, m_velocities.size()-2);
+    std::vector<Eigen::Triplet<double> > A_triplets;
```

Comment:
I don't know if this part is a computational bottleneck, but if this is  the case we could move this objects to be attributes of the class.

Reply:
I thought not to put it in the class definition for a couple of reasons:
- I did not want to include ``Eigen`` in the header
- The matrix is set from triplets at every call. Those triplets depends upon the user defined data, thus they would have been deleted and overwritten in any case
- This method is called only when new data is set, in principle it should not be frequent.

In any case if it become a bottleneck I agree that we should think about a smarter way to define the triplets and generate the sparse matrix.